packages/python/src/components/DataclassDeclaration.tsx

```diff
@@ -0,0 +1,189 @@
+import {
+  For,
+  Show,
+  childrenArray,
+  computed,
+  findKeyedChildren,
+  taggedComponent,
+} from "@alloy-js/core";
+import { dataclassesModule } from "../builtins/python.js";
+import { findMethodDeclaration } from "../utils.js";
+import { Atom } from "./Atom.jsx";
+import type { ClassDeclarationProps } from "./ClassDeclaration.js";
+import { ClassDeclaration } from "./ClassDeclaration.js";
+import { StatementList } from "./StatementList.js";
+
+/**
+ * Validate the keyword arguments for the Python `@dataclass(...)` decorator.
+ */
+function validateDataclassDecoratorKwargs(
+  children: any[],
+  kwargs: DataclassDecoratorKwargs,
+) {
+  if (kwargs.weakref_slot === true && kwargs.slots !== true) {
+    throw new Error(
+      "weakref_slot=True requires slots=True in @dataclass decorator",
+    );
+  }
+
+  if (kwargs.order === true && kwargs.eq === false) {
+    throw new Error("order=True requires eq=True in @dataclass decorator");
+  }
+
+  if (kwargs.order === true) {
+    const orderingMethods = ["__lt__", "__le__", "__gt__", "__ge__"];
+    const conflict = findMethodDeclaration(children, orderingMethods);
```

Comment:
I'm not sure this validation is necessary, especially given that this will only find method declarations if they are present as direct children of this component. If I have e.g. a set of methods bundled into some reusable component, this won't find it. I'd say if you want to keep this kind of validation it needs to operate at the symbol level (checking to see if a symbol for this method name already exists).